src/OpenTelemetry.Exporter.Jaeger/OpenTelemetry.Exporter.Jaeger.csproj

```diff
@@ -24,8 +25,15 @@
 
   <ItemGroup>
     <Reference Include="ApacheThrift">
+      <!-- Special care is given to the ApacheThrift reference because the package contains a net45 binary with a breaking API. Be careful modifying this. -->
       <HintPath>$(PkgApacheThrift)\lib\netstandard2.0\Thrift.dll</HintPath>
     </Reference>
   </ItemGroup>
 
+  <Target Name="CopyReferencesToPackage">
+    <ItemGroup>
+      <TfmSpecificPackageFile Include="$(OutputPath)\Thrift.dll" PackagePath="lib/$(TargetFramework)" />

```

Comment:
is it needed to be copied on all target FW? Should it come from the NuGet?

Reply:
The Apache Thrift NuGet is bugged, IMO. If we referenced ApacheThrift the normal way this is what would happen:

The OT Jaeger lib is targeting .NET Standard 2.0/2.1. It compiles against the ApacheThrift /lib/netstandard2.0 API and everything works great.

Now a .NET Framework 4.8 application references our Jaeger library. It will get the .NET Standard 2.0 OT Jaeger binary from NuGet. Then it will get the ApacheThrift net45 binary from NuGet. That binary has a completely different API surface than the one our library compiled against so now the application won't build.

All of this csproj voodoo is to work around the problem of ApacheThrift shipping different APIs in the same NuGet package and make it so the Apache Thrift binary resolves correctly to the .NET Standard 2.0 version at the end of the day.